### PR TITLE
Add recommendation to store sensitive data (API keys and URLs, tracki…

### PR DESCRIPTION
**Pull request by iverc**
…ng id's, etc) in build.gradle & grade.properties

By storing sensitive data (API keys and URLs, tracking ids, etc) in build.gradle & gradle.properties files instead of config.xml file that can be easily decompiled we ensure that sensitive data is stored securely and will not be compromised. Additional advantage is that you can access those constants with BuildConfig.GOOGLE_API_KEY where Context is not available.

### CODE REVIEW

**Comment by iverc** [comment]
PTAL

File: README.md [hunk 1]
```diff
@@ -135,6 +135,56 @@ signingConfigs {
 }
 ```
 
++**Store sensitive data (API keys & urls, tracking ids, etc) in build.gradle & gradle.properties files** instead of config.xml file that can be easily decompiled. Otherwise your data is compromised.
+Additional advantage is that you can access those constants with BuildConfig.GOOGLE_API_KEY where Context is not available. 
+
+First, add constants into a `gradle.properties` file which should _not_ be added to the version control system:
```
Comment (piotrek1543) [inline README.md]: `_not_` looks strange, maybe capitalizing letters or bolding would be enough ;-)

Comment (piotrek1543) [inline README.md]: `_not_` looks strange, maybe capitalizing letters or making this bold would be enough ;-)

Comment (iverc) [inline README.md]: I followed an example of previous section about passwords. If _not_ does not fit, both sections should be updated ;-)

**Comment by iverc** [comment]
Fixed the _not_ issue, PTAL

**Comment by rpattabi** [comment]
Isn't BuildConfig.java easily available through APK? How is this more secure?

**Comment by iverc** [comment]
Actually its not. Tested with dex2jar/jadx. And all xml files including manifest are available with simple apktool.

**Comment by iverc** [comment]
PTAL

**Comment by peter-tackage** [comment]
Using `gradle.properties` is already our recommendation for providing app "secrets". 

By including them in BuildConfig by doing something like ` buildConfigField(STRING, "FREESOUND_API_URL", freesound_api_url)` *does* make them visible if the APK is deconstructed (I've just done this with [jadx](https://github.com/skylot/jadx/). 

The benefit of the technique is that you can avoid having the the "secrets"  in your repo, which wouldn't be the case if you directly defined them in a String resources file.

**Comment by pedpess** [comment]
Shall we close this PR?  @peter-tackage

**Pull request closed without merging**